backend/program/libaries/symlink.py

```diff
@@ -1,85 +1,94 @@
 from typing import Generator
 
-from utils.logger import logger
+from program.media.item import Episode, MediaItem, Movie, Season, Show
 from program.settings.manager import settings_manager
-from program.media.item import (
-    MediaItem,
-    Movie,
-    Show,
-    Season,
-    Episode,
-)
+from utils.logger import logger
+
 
 class SymlinkLibrary:
     def __init__(self):
         self.key = "symlinklibrary"
         self.last_fetch_times = {}
         self.settings = settings_manager.settings.symlink
-        self.initialized = True
         self.initialized = self.validate()
         if not self.initialized:
-            logger.error("SymlinkLibrary initialization failed due to invalid configuration.")
+            logger.error(
+                "SymlinkLibrary initialization failed due to invalid configuration."
+            )
             return
-    
+
     def validate(self) -> bool:
-        status = all(
-            os.path.exists(self.settings.library_path / d) 
-            for d in ("shows", "movies")
-        )
-        if not status:
-            folders = os.listdir(self.settings.library_path)
-            logger.debug("library dir contains %s", ", ".join(folders))
-        return status
-    
+        library_path = Path(self.settings.library_path).resolve()
+        # We don't want to allow the library path to be the current directory
+        if library_path == Path.cwd().resolve():
+            logger.error(
+                "Library path not set or set to the current directory in SymlinkLibrary settings."
+            )
+            return False
+
+        required_dirs = ["shows", "movies"]
+        missing_dirs = [d for d in required_dirs if not (library_path / d).exists()]
+
+        if missing_dirs:
+            available_dirs = ", ".join(os.listdir(library_path))
+            logger.error(
+                f"Missing required directories in the library path: {', '.join(missing_dirs)}."
+            )
+            logger.debug(f"Library directory contains: {available_dirs}")
+            return False
+        return True
+
     def run(self) -> Generator[MediaItem, None, None]:
         """Create a library from the symlink paths.  Return stub items that should
         be fed into an Indexer to have the rest of the metadata filled in."""
         movies = [
-            (root, files[0]) 
-            for root, _, files 
-            in os.walk(self.settings.library_path / "movies") 
+            (root, files[0])
+            for root, _, files in os.walk(self.settings.library_path / "movies")
             if files
         ]
         for path, filename in movies:
-            imdb_id = re.search('(tt\d+)', filename)
+            imdb_id = re.search("(tt\d+)", filename)
             if not imdb_id:
-                logger.error("Can't extract episode imdb_id at path %s", path / filename)
+                logger.error(
+                    "Can't extract episode imdb_id at path %s", path / filename
```

Comment:
"Can't extract **movie** imdb_id at path %s"

Reply:
fixed in next commit.